src/Compilers/Core/Portable/Emit/EditAndContinue/EmitDifferenceResult.cs

```diff
@@ -8,9 +8,9 @@ namespace Microsoft.CodeAnalysis.Emit
 {
     public sealed class EmitDifferenceResult : EmitResult
     {
-        public EmitBaseline Baseline { get; }
+        public EmitBaseline? Baseline { get; }
 
-        internal EmitDifferenceResult(bool success, ImmutableArray<Diagnostic> diagnostics, EmitBaseline baseline) :
+        internal EmitDifferenceResult(bool success, ImmutableArray<Diagnostic> diagnostics, EmitBaseline? baseline) :
             base(success, diagnostics)
         {
             Baseline = baseline;
```

Comment:
Should there be a member introduced like `[MemberNotNullWhen(true, nameof(Baseline))] public new bool Success => base.Success;`?

Reply:
Good point. I'll add it.

---
In reply to: [585787947](https://github.com/dotnet/roslyn/pull/51576#discussion_r585787947) [](ancestors = 585787947)

Reply:
Actually, I'm not sure if we should add a new public member just so we can annotate Baseline. it'd be useful to have an attribute that can be applied to Baseline that says NonNullWhenMember(nameof(Success), true)

---
In reply to: [585906269](https://github.com/dotnet/roslyn/pull/51576#discussion_r585906269) [](ancestors = 585906269,585787947)

Reply:
I think the ability to "adjust" nullable annotations of a base type member from a derived type, without having to "truly" override or shadow it, is something we've looked at in nullable before. Not sure what we will end up doing to address the scenario.